python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1480,43 +1488,45 @@ def _get_partition_time_windows_not_in_subset(
         Each time window is a single partition.
         """
         first_tw = cast(
-            TimeWindowPartitionsDefinition, self.partitions_def
+            TimeWindowPartitionsDefinition, self.get_partitions_def()
         ).get_first_partition_window(current_time=current_time)
         last_tw = cast(
-            TimeWindowPartitionsDefinition, self.partitions_def
+            TimeWindowPartitionsDefinition, self.get_partitions_def()
         ).get_last_partition_window(current_time=current_time)
 
         if not first_tw or not last_tw:
             check.failed("No partitions found")
 
-        if len(self.included_time_windows) == 0:
+        if len(self.get_included_time_windows()) == 0:
```

Comment:
would be cleaner in this method if there was a single `included_time_windows = self.get_included_time_windows()` up here, rather than invoking `get_included_time_windows()` a bunch below